### PR TITLE
feat: redpacket history

### PR DESCRIPTION
**Pull request by beyond009**
## Description

Closes # (NO_ISSUE)
MF-0000
https://mask.atlassian.net/browse/FW-279
## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  theme,
+  { listItemBackground, listItemBackgroundIcon },
+) => {
+  const smallQuery = `@media (max-width: ${theme.breakpoints.values.sm}px)`
+  return {
+    message: {
+      whiteSpace: 'nowrap',
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      [smallQuery]: {
```
Comment (UncleBill) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx]: Looks like you bypass the git commit hook, code indent width should be 4 spaces instead of 2 ones. Follow our code style, you can format with prettier.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                  <TokenIcon
+                    className={classes.icon}
+                    address={''}
+                    name={token_symbol}
+                    logoURL={token_logo}
+                  />
+                  : null}
+              </div>
+              <button className={classes.moreDetails} onClick={() => { handleOpenDetails(redpacket_id) }}>
+                More details
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]: Locales

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                  </Typography>
+                </div>
+                <div className={classes.fullWidthBox}>
+                  <Typography variant="body1" className={cx(classes.infoTitle, classes.message)}>
+                    {t.received_time()}
+                  </Typography>
+                  <Typography
+                    variant="body1"
+                    className={cx(classes.info, classes.message)}>
+                    {t.history_duration({ time: dateTimeFormat(new Date(Number(received_time) * 1000)) })}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]: Use `date-fns`.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+  return (
+    <ListItem className={classes.root}>
+      <section className={classes.contentItem}>
+        <Box className={classes.box}>
+          <Box className={classes.content}>
+            <section className={classes.section}>
+              <div className={classes.div}>
+                <div className={classes.fullWidthBox}>
+                  <Typography variant="body1" className={cx(classes.title, classes.message)}>
+                    {rp_msg === '' ?
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]:
```suggestion
                    {!rp_msg ?
```

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+                          { significant: 2, isPrecise: true },
+                        ),
+                      symbol: token_symbol,
+                    }}
+                  />
+                </Typography>
+                {token_logo ?
+                  <TokenIcon
+                    className={classes.icon}
+                    address={''}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx]: It should be a valid EVM address.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+                  </Typography>
+                </div>
+                <div className={classes.fullWidthBox}>
+                  <Typography variant="body1" className={cx(classes.infoTitle, classes.message)}>
+                    {t.create_time()}
+                  </Typography>
+                  <Typography
+                    variant="body1"
+                    className={cx(classes.info, classes.message, redpacket_id ? '' : classes.invisible)}>
+                    {t.history_duration({ time: dateTimeFormat(new Date(Number(create_time) * 1000)) })}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx]: Use `date-fns`.

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+                  <TokenIcon
+                    className={classes.icon}
+                    address={''}
+                    name={token_symbol}
+                    logoURL={token_logo}
+                  />
+                  : null}
+              </div>
+              <button className={classes.moreDetails} onClick={() => { handleOpenDetails(redpacket_id) }}>
+                More details
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx]: Locales

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+                  </Typography>
+                </div>
+                <div className={classes.fullWidthBox}>
+                  <Typography variant="body1" className={cx(classes.infoTitle, classes.message)}>
+                    {t.create_time()}
+                  </Typography>
+                  <Typography
+                    variant="body1"
+                    className={cx(classes.info, classes.message, redpacket_id ? '' : classes.invisible)}>
+                    {t.history_duration({ time: dateTimeFormat(new Date(Number(create_time) * 1000)) })}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx]: Use `date-fns`.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx (hunk 5 above)
Comment (beyond009) [reply packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx]: Data source does not contain token address. Address is not required when we have the token logo url

File: packages/shared/src/UI/components/TokenIcon/index.tsx [hunk 9]
```diff
@@ -44,7 +44,7 @@ export const TokenIcon = memo(function TokenIcon(props: TokenIconProps) {
                             "Unreachable: it used to call Hub.getNonFungibleTokenIconURLs, but it's default implementation is throw an error, and not override by any subclass.",
                         )
                     })()
-                :   await Hub.getFungibleTokenIconURLs(chainId, address).catch(() => [])
```
Comment (Jack-Works) [inline packages/shared/src/UI/components/TokenIcon/index.tsx]: do you have the correct prettier version installed?

File: pnpm-lock.yaml [hunk 10]
```diff
@@ -7792,7 +7792,7 @@ packages:
     resolution: {integrity: sha512-yNYzYN6N2wrcoOcxfNgBP7GLVoewfV7vsg4muVLAJdIBQWnXx+t7d/MEp0n3BYhcxH4MEjm6IZ6jDVzFa3zkJw==}
     engines: {node: '>= 12'}
     peerDependencies:
-      webpack: Jack-Works/webpack#528c91e564d5756e21c9c462b607d913452af770
```
Comment (Jack-Works) [inline pnpm-lock.yaml]: this lock file change looks incorrect. this PR does not really need to change lock file, please revert this change.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+        },
+        moreDetails: {
+            fontSize: 12,
+            fontStyle: 'normal',
+            fontWeight: 700,
+            lineHeight: '16px',
+            background: 'none',
+            cursor: 'pointer',
+            border: 'none',
+            color: '#181818',
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]: Use `secondaryMain` from MaskColor.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+            width: '100%',
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'space-between',
+            flexWrap: 'nowrap',
+            marginTop: 15,
+        },
+        footerInfo: {
+            fontSize: 14,
+            color: '#181818',
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]: Use `secondaryMain` from MaskColor.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketHistroyList.tsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        data: historiesData,
+        isLoading,
+        fetchNextPage,
+    } = useSuspenseInfiniteQuery({
+        queryKey: ['fireflyRedPacketHistory', account, historyType],
+        initialPageParam: createIndicator(undefined, ''),
+        queryFn: async ({ pageParam }) => {
+            const res = await FireflyRedPacket.getHistory(
+                historyType,
+                '0x790116d0685eb197b886dacad9c247f785987a4a',
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketHistroyList.tsx]: Use the current connect wallet.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketHistroyDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+        queryKey: ['fireflyClaimHistory', rpid],
+        initialPageParam: '',
+        queryFn: async ({ pageParam }) => {
+            const res = await FireflyRedPacket.getClaimHistory(rpid, createIndicator(undefined, pageParam as string))
+            return res
+        },
+        getNextPageParam: (lastPage) => lastPage.cursor,
+    })
+    const claimList = useMemo(() => claimData?.pages.flatMap((x) => x.list) ?? [], [claimData])
+    const claimInfo = useMemo(() => first(claimData.pages), [claimData])
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketHistroyDetails.tsx]: Merge into signle memo.

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
+                                    </Typography>
+                                    <Typography
+                                        variant="body1"
+                                        className={cx(
+                                            classes.info,
+                                            classes.message,
+                                            redpacket_id ? '' : classes.invisible,
+                                        )}>
+                                        {t.history_duration({
+                                            time: dateTimeFormat(new Date(Number(create_time) * 1000)),
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx]:
```suggestion
                                            time: dateTimeFormat(fromUnixTime(Number.parseInt(create_time, 10))),
```

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 16]
```diff
@@ -0,0 +1,257 @@
...
+                                    </Typography>
+                                    <Typography
+                                        variant="body1"
+                                        className={cx(
+                                            classes.info,
+                                            classes.message,
+                                            redpacket_id ? '' : classes.invisible,
+                                        )}>
+                                        {t.history_duration({
+                                            time: format(new Date(Number(create_time) * 1000), 'M/d/yyyy HH:mm'),
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx]:
```suggestion
                                            time: format(fromUnixTime(Number.parseInt(create_time, 10)), 'M/d/yyyy HH:mm'),
```

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+                                        {!rp_msg ? t.best_wishes() : rp_msg}
+                                    </Typography>
+                                </div>
+                                <div className={classes.fullWidthBox}>
+                                    <Typography variant="body1" className={cx(classes.infoTitle, classes.message)}>
+                                        {t.received_time()}
+                                    </Typography>
+                                    <Typography variant="body1" className={cx(classes.info, classes.message)}>
+                                        {t.history_duration({
+                                            time: format(new Date(Number(received_time) * 1000), 'M/d/yyyy HH:mm'),
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]:
```suggestion
                                            time: format(fromUintTime(Number.parseInt(received_time, 10)), 'M/d/yyyy HH:mm'),
```

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+        },
+        moreDetails: {
+            fontSize: 12,
+            fontStyle: 'normal',
+            fontWeight: 700,
+            lineHeight: '16px',
+            background: 'none',
+            cursor: 'pointer',
+            border: 'none',
+            color: '#181818',
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx]: secondaryMain

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 19]
```diff
@@ -0,0 +1,257 @@
...
+    return (
+        <ListItem className={classes.root}>
+            <section className={classes.contentItem}>
+                <Box className={classes.box}>
+                    <Box className={classes.content}>
+                        <section className={classes.section}>
+                            <div className={classes.div}>
+                                <div className={classes.fullWidthBox}>
+                                    <Typography variant="body1" className={cx(classes.title, classes.message)}>
+                                        {rp_msg === '' ? t.best_wishes() : rp_msg}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx:187]:
```suggestion
                                        {!rp_msg ? t.best_wishes() : rp_msg}
```

File: packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx [hunk 20]
```diff
@@ -0,0 +1,257 @@
...
+
+    return (
+        <ListItem className={classes.root}>
+            <section className={classes.contentItem}>
+                <Box className={classes.box}>
+                    <Box className={classes.content}>
+                        <section className={classes.section}>
+                            <div className={classes.div}>
+                                <div className={classes.fullWidthBox}>
+                                    <Typography variant="body1" className={cx(classes.title, classes.message)}>
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FIreflyRedPacketInSentHistoryList.tsx]: DRY. Use `<FireflyRedPacketDetailsItem />`

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx [hunk 21]
```diff
@@ -0,0 +1,242 @@
...
+    return (
+        <ListItem className={classes.root}>
+            <section className={classes.contentItem}>
+                <Box className={classes.box}>
+                    <Box className={classes.content}>
+                        <section className={classes.section}>
+                            <div className={classes.div}>
+                                <div className={classes.fullWidthBox}>
+                                    <Typography variant="body1" className={cx(classes.title, classes.message)}>
+                                        {rp_msg === '' ? t.best_wishes() : rp_msg}
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketDetailsItem.tsx]:
```suggestion
                                        {!rp_msg ? t.best_wishes() : rp_msg}
```

File: packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx [hunk 22]
```diff
@@ -0,0 +1,241 @@
...
+
+    return (
+        <ListItem className={classes.root}>
+            <section className={classes.contentItem}>
+                <Box className={classes.box}>
+                    <Box className={classes.content}>
+                        <section className={classes.section}>
+                            <div className={classes.div}>
+                                <div className={classes.fullWidthBox}>
+                                    <Typography variant="body1" className={cx(classes.title, classes.message)}>
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketInClaimedHistoryList.tsx]: DRY

**Review by guanbinrui — APPROVED** [review]
